src/algo/maximum_flow/ford_fulkerson.rs

```diff
@@ -0,0 +1,215 @@
+{
+    if vertex == edge.source() {
+        // backward edge
+        flow
+    } else if vertex == edge.target() {
+        // forward edge
+        *edge.weight() - flow
+    } else {
+        let end_point = NodeIndexable::to_index(&network, vertex);
+        panic!("Illegal endpoint {}", end_point);
+    }
+}
+
+/// Gets the other endpoint of graph edge, if any, otherwise panics.
+fn other_endpoint<G>(network: G, edge: G::EdgeRef, vertex: G::NodeId) -> G::NodeId
+where
+    G: NodeIndexable + IntoEdges,
+{
+    if vertex == edge.source() {
+        edge.target()
+    } else if vertex == edge.target() {
+        edge.source()
+    } else {
+        let end_point = NodeIndexable::to_index(&network, vertex);
+        panic!("Illegal endpoint {}", end_point);
+    }
+}
+
+/// Tells whether there is an augmented path in the graph
+fn has_augmented_path<G>(
+    network: G,
+    source: G::NodeId,
+    destination: G::NodeId,
+    edge_to: &mut [Option<G::EdgeRef>],
+    flows: &[G::EdgeWeight],
+) -> bool
+where
+    G: NodeCount + IntoEdgesDirected + NodeIndexable + EdgeIndexable + Visitable,
+    G::EdgeWeight: Sub<Output = G::EdgeWeight> + PositiveMeasure,
+{
+    let mut visited = network.visit_map();
+    let mut queue = VecDeque::new();
+    visited.visit(source);
+    queue.push_back(source);
+
+    while let Some(vertex) = queue.pop_front() {
+        let out_edges = network.edges_directed(vertex, Direction::Outgoing);
+        let in_edges = network.edges_directed(vertex, Direction::Incoming);
+        for edge in out_edges.chain(in_edges) {
+            let next = other_endpoint(&network, edge, vertex);
+            let edge_index: usize = EdgeIndexable::to_index(&network, edge.id());
+            let residual_cap = residual_capacity(&network, edge, next, flows[edge_index]);
+            if !visited.is_visited(&next) && (residual_cap > G::EdgeWeight::zero()) {
+                visited.visit(next);
+                edge_to[NodeIndexable::to_index(&network, next)] = Some(edge);
+                if destination == next {
+                    return true;
+                }
+                queue.push_back(next);
+            }
+        }
+    }
+    false
+}
+
+fn adjust_residual_flow<G>(
+    network: G,
+    edge: G::EdgeRef,
+    vertex: G::NodeId,
+    flow: G::EdgeWeight,
+    delta: G::EdgeWeight,
+) -> G::EdgeWeight
+where
+    G: NodeIndexable + IntoEdges,
+    G::EdgeWeight: Sub<Output = G::EdgeWeight> + PositiveMeasure,
+{
+    if vertex == edge.source() {
+        // backward edge
+        flow - delta
+    } else if vertex == edge.target() {
+        // forward edge
+        flow + delta
+    } else {
+        let end_point = NodeIndexable::to_index(&network, vertex);
+        panic!("Illegal endpoint {}", end_point);
+    }
+}
+
+/// [Ford-Fulkerson][ff] algorithm in the [Edmonds-Karp][ek] variation.
+///
+/// Computes the [maximum flow] from `source` to `destination` in a weighted directed graph.
```

Comment:
```suggestion
/// [Ford-Fulkerson][ff] algorithm in the [Edmonds-Karp][ek] variation.
/// Computes the [maximum flow] from `source` to `destination` in a weighted directed graph.
```
This makes it such that on the module docs for `maximum_flow`, the description of Ford Fulkerson also shows the text about computing a max flow :)